src/main/java/org/folio/service/protection/AcquisitionsUnitsServiceImpl.java

```diff
@@ -28,92 +29,105 @@ public class AcquisitionsUnitsServiceImpl extends BaseService implements Acquisi
+  private Future<List<String>> getAcqUnitIdsForSearch(String lang, Context context, Map<String, String> headers) {
+    Promise<List<String>> promise = Promise.promise();
     return getAcqUnitIdsForUser(headers.get(OKAPI_USERID_HEADER), lang, context, headers)
-      .thenCombine(getOpenForReadAcqUnitIds(lang, context, headers), (unitsForUser, unitsAllowRead) -> StreamEx.of(unitsForUser, unitsAllowRead)
-        .flatCollection(strings -> strings)
-        .distinct()
-        .toList());
+      .compose(unitsForUser -> getOpenForReadAcqUnitIds(lang, context, headers).compose(unitsAllowRead -> {
+        List<String> list =  StreamEx.of(unitsForUser, unitsAllowRead)
+          .flatCollection(strings -> strings)
+          .distinct()
+          .toList();
+        promise.complete(list);
+        return promise.future();
+      }));
   }
 
-  private CompletableFuture<List<String>> getAcqUnitIdsForUser(String userId, String lang, Context context, Map<String, String> headers) {
+  private Future<List<String>> getAcqUnitIdsForUser(String userId, String lang, Context context, Map<String, String> headers) {
     logger.debug("getAcqUnitIdsForUser:: Trying to get acquisition unit ids with userId: {}", userId);
+    Promise<List<String>> promise = Promise.promise();
     return getAcquisitionsUnitsMemberships("userId==" + userId, 0, Integer.MAX_VALUE, lang, context, headers)
-      .thenApply(memberships -> {
+      .compose(memberships -> {
         List<String> ids = memberships.getAcquisitionsUnitMemberships()
           .stream()
           .map(AcquisitionsUnitMembership::getAcquisitionsUnitId)
           .collect(Collectors.toList());
-
-        if (logger.isDebugEnabled()) {
           logger.debug("getAcqUnitIdsForUser:: User belongs to {} acq units: {}", ids.size(), StreamEx.of(ids).joining(", "));
-        }
-
-        return ids;
+        promise.complete(ids);
+        return promise.future();
       });
   }
 
-  private CompletableFuture<List<String>> getOpenForReadAcqUnitIds(String lang, Context context, Map<String, String> headers) {
+  private Future<List<String>> getOpenForReadAcqUnitIds(String lang, Context context, Map<String, String> headers) {
     logger.debug("getOpenForReadAcqUnitIds:: Trying to get acquisition unit ids with open status");
+    Promise<List<String>> promise = Promise.promise();
     return getAcquisitionsUnits("protectRead==false", 0, Integer.MAX_VALUE, lang, context, headers)
-      .thenApply(units -> {
+      .compose(units -> {
         List<String> ids = units.getAcquisitionsUnits()
           .stream()
           .map(AcquisitionsUnit::getId)
           .collect(Collectors.toList());
         if (logger.isDebugEnabled()) {
           logger.debug("getOpenForReadAcqUnitIds:: {} acq units with 'protectRead==false' are found: {}", ids.size(), StreamEx.of(ids).joining(", "));
         }
-        return ids;
+        promise.complete(ids);
```

Comment:
Promise is not needed here, just return list of ids

Reply:
fixed